File: 20220808/112.py
from typing import Optional
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def hasPathSum(self, root: Optional[TreeNode], targetSum: int) -> bool:
        if root == None : return False
        targetSum -= root.val
        result = False
        left_result = False
        right_result = False
        if root.left == None and root.right == None:
            result = (targetSum == 0)
        else:
            left_result = self.hasPathSum(root.left, targetSum)
            right_result = self.hasPathSum(root.right, targetSum)
        result = result or left_result or right_result
        return result
s = Solution()
print(s.hasPathSum(root = TreeNode(val = 1, left = TreeNode(val=2)), targetSum = 0))

File: 20221009/112.cc
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
//Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    bool hasPathSum(TreeNode* root, int targetSum) {
        
    }
};
int main(void){
    vector<int> test = {1,2,3};
    vector<int> test2(4);
    find(test.begin(), test.end(), 3);
    cout<<"hello world"<<endl;
    return 0;
}
